orchagent/fdborch.cpp

```diff
@@ -529,23 +816,197 @@ bool FdbOrch::removeFdbEntry(const FdbEntry& entry)
     status = sai_fdb_api->remove_fdb_entry(&fdb_entry);
     if (status != SAI_STATUS_SUCCESS)
     {
-        SWSS_LOG_ERROR("Failed to remove FDB entry. mac=%s, bv_id=0x%" PRIx64,
+        SWSS_LOG_ERROR("FdbOrch RemoveFDBEntry: Failed to remove FDB entry. mac=%s, bv_id=0x%lx",
                        entry.mac.to_string().c_str(), entry.bv_id);
-        return true; //FIXME: it should be based on status. Some could be retried. some not
+        return false; //FIXME: it should be based on status. Some could be retried. some not
     }
-
+    port.m_fdb_count--;
+    m_portsOrch->setPort(port.m_alias, port);
+    vlan.m_fdb_count--;
+    m_portsOrch->setPort(vlan.m_alias, vlan);
     (void)m_entries.erase(entry);
 
-    gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_FDB_ENTRY);
+    // Remove in StateDb
+    m_fdbStateTable.del(key);
 
-    Port port;
-    m_portsOrch->getPortByBridgePortId(entry.bv_id, port);
+    gCrmOrch->decCrmResUsedCounter(CrmResourceType::CRM_FDB_ENTRY);
 
-    FdbUpdate update = {entry, port, false};
+    FdbUpdate update = {entry, port, fdbData.type, false};
     for (auto observer: m_observers)
     {
         observer->update(SUBJECT_TYPE_FDB_CHANGE, &update);
     }
 
     return true;
 }
+
+int FdbOrch::flushFdbAll(bool flush_static)
+{
+    sai_status_t status;
+    sai_attribute_t port_attr;
+
+    if (!flush_static)
+    {
+        port_attr.id = SAI_FDB_FLUSH_ATTR_ENTRY_TYPE;
+        port_attr.value.s32 = SAI_FDB_FLUSH_ENTRY_TYPE_DYNAMIC;
+        status = sai_fdb_api->flush_fdb_entries(gSwitchId, 1, &port_attr);
+    }
+    else
+    {
+        status = sai_fdb_api->flush_fdb_entries(gSwitchId, 0, NULL);
+    }
+    if (status != SAI_STATUS_SUCCESS)
+    {
+        SWSS_LOG_ERROR("Flush fdb failed, return code %x", status);
+    }
+    return 0;
+}
+
+int FdbOrch::flushFdbByPort(const string &alias, bool flush_static)
+{
+    sai_status_t status;
+    Port port;
+    sai_attribute_t port_attr[2];
+
+    if (!m_portsOrch->getPort(alias, port))
+    {
+        SWSS_LOG_ERROR("could not locate port from alias %s", alias.c_str());
+        return false;
```

Comment:
this is not a bool function, you should return error code.

Reply:
will change function to return bool

Reply:
done